string filename
string name
string description
int32 level
bool custom
